### PR TITLE
Move legacy transformations and ops to legacy library

### PR DESCRIPTION
**Pull request by GlebKazantaev**
In this PR I move all legacy transformations to legacy_api. Some of transformations (ConvertConvolutoins and ConvBiasFusion) and internal ngraph ops were left inside transformations library as a common part used by plugins. List of common ops/transformations can be increased as it depends on plugins needs.
Also this separation help to decrease inference_engine_transformation.so size from 3.3M to 2.2M.
Structure below shows new folders structure. I intentionally show only hpp files location as cpp files has the same structure.

```
.
├── legacy_api
│   └── include
│       └── legacy
│           ├── ngraph_ops <--- all legacy ops
│           └── transformations <--- all legacy transformations
│               └── convert_opset1_to_legacy
└── transformations
    └── include
        ├── ngraph_ops <--- videly used ops that are allowed for common use
        └── transformations
            ├── common_optimizations <--- only optimization passes (fusions)
            ├── control_flow <--- passes related to cycles
            ├── op_conversions <--- decomposition/conversion passes
            ├── opset_conversions <--- convertOpSetXToOpSetY passes
            ├── rt_info
            ├── smart_reshape
            └── utils
```

**Detailed structue:**
```
.
├── legacy_api
│   └── include
│       └── legacy
│           ├── ngraph_ops
│           │   ├── crop_ie.hpp
│           │   ├── eltwise.hpp
│           │   ├── fully_connected.hpp
│           │   ├── gather_ie.hpp
│           │   ├── gather_tree_ie.hpp
│           │   ├── gru_cell_ie.hpp
│           │   ├── gru_sequence_ie.hpp
│           │   ├── hard_sigmoid_ie.hpp
│           │   ├── interp.hpp
│           │   ├── lrn_ie.hpp
│           │   ├── lstm_cell_ie.hpp
│           │   ├── lstm_sequence_ie.hpp
│           │   ├── nms_ie.hpp
│           │   ├── normalize_ie.hpp
│           │   ├── onehot_ie.hpp
│           │   ├── pad_ie.hpp
│           │   ├── power.hpp
│           │   ├── prior_box_clustered_ie.hpp
│           │   ├── prior_box_ie.hpp
│           │   ├── proposal_ie.hpp
│           │   ├── relu_ie.hpp
│           │   ├── rnn_cell_ie.hpp
│           │   ├── rnn_sequence_ie.hpp
│           │   ├── scaleshift.hpp
│           │   ├── selu_ie.hpp
│           │   ├── swish_ie.hpp
│           │   ├── tile_ie.hpp
│           │   └── topk_ie.hpp
│           └── transformations
│               └── convert_opset1_to_legacy
│                   ├── convert_cells_to_cells_ie.hpp
│                   ├── convert_gather_to_gather_ie.hpp
│                   ├── convert_gathertree_to_gathertree_ie.hpp
│                   ├── convert_hard_sigmoid_to_hard_sigmoid_ie.hpp
│                   ├── convert_interpolate_to_interp_or_resample.hpp
│                   ├── convert_lrn_to_lrn_ie.hpp
│                   ├── convert_matmul_to_fc_or_gemm.hpp
│                   ├── convert_mul_add_to_scaleshift_or_power.hpp
│                   ├── convert_mul_or_add_finally.hpp
│                   ├── convert_nms_4_to_legacy.hpp
│                   ├── convert_nms_to_nms_ie.hpp
│                   ├── convert_normalizel2_to_normalize_ie.hpp
│                   ├── convert_one_hot_to_one_hot_ie.hpp
│                   ├── convert_opset1_to_legacy.hpp
│                   ├── convert_pad_to_pad_ie.hpp
│                   ├── convert_power_to_power_ie.hpp
│                   ├── convert_prelu_to_relu_ie.hpp
│                   ├── convert_prior_to_ie_prior.hpp
│                   ├── convert_proposal_to_proposal_ie.hpp
│                   ├── convert_selu_to_selu_ie.hpp
│                   ├── convert_sequences_to_sequences_ie.hpp
│                   ├── convert_sqrt_to_power_ie.hpp
│                   ├── convert_strided_slice_to_crop.hpp
│                   ├── convert_swish_to_swish_ie.hpp
│                   ├── convert_tile_to_ie_tile.hpp
│                   ├── convert_topk_to_topk_ie.hpp
│                   ├── fc_bias_fusion.hpp
│                   ├── reshape_1d_ops.hpp
│                   ├── reshape_fc_fusion.hpp
│                   └── reshape_fully_connected.hpp
└── transformations
    └── include
        ├── ngraph_ops
        │   ├── convolution_ie.hpp
        │   ├── deconvolution_ie.hpp
        │   └── type_relaxed.hpp
        └── transformations
            ├── common_optimizations
            │   ├── algebraic_simplification.hpp
            │   ├── common_optimizations.hpp
            │   ├── conv_bias_fusion.hpp
            │   ├── conv_mul_fusion.hpp
            │   ├── depth_to_space_fusion.hpp
            │   ├── fq_mul_fusion.hpp
            │   ├── fq_reshape_fusion.hpp
            │   ├── hswish_fusion.hpp
            │   ├── lin_op_sequence_fusoin.hpp
            │   ├── mish_fusion.hpp
            │   ├── nop_elimination.hpp
            │   ├── normalize_l2_fusion.hpp
            │   ├── optimize_strided_slice.hpp
            │   ├── pull_transpose_through_fq.hpp
            │   ├── remove_filtering_boxes_by_size.hpp
            │   ├── softplus_fusion.hpp
            │   ├── softplus_to_mish_fusion.hpp
            │   └── swish_fusion.hpp
            ├── control_flow
            │   └── unroll_tensor_iterator.hpp
            ├── op_conversions
            │   ├── batch_norm_decomposition.hpp
            │   ├── bidirectional_sequences_decomposition.hpp
            │   ├── convert_batch_to_space.hpp
            │   ├── convert_broadcast3.hpp
            │   ├── convert_broadcast_to_tiles.hpp
            │   ├── convert_convolutions.hpp
            │   ├── convert_depth_to_space.hpp
            │   ├── convert_divide.hpp
            │   ├── convert_extract_image_patches_to_reorg_yolo.hpp
            │   ├── convert_gelu.hpp
            │   ├── convert_minimum_to_power_and_max.hpp
            │   ├── convert_mod.hpp
            │   ├── convert_negative.hpp
            │   ├── convert_nms3.hpp
            │   ├── convert_pad_to_group_conv.hpp
            │   ├── convert_quantize_dequantize.hpp
            │   ├── convert_reduce_to_pooling.hpp
            │   ├── convert_scatter_elements_to_scatter.hpp
            │   ├── convert_shapeof3.hpp
            │   ├── convert_shuffle_channels3.hpp
            │   ├── convert_space_to_batch.hpp
            │   ├── convert_space_to_depth.hpp
            │   ├── convert_subtract.hpp
            │   ├── convert_ti_to_sequences.hpp
            │   ├── convert_topk3.hpp
            │   ├── gru_cell_decomposition.hpp
            │   ├── hswish_decomposition.hpp
            │   ├── lstm_cell_decomposition.hpp
            │   ├── reduce_l1_decomposition.hpp
            │   ├── reduce_l2_decomposition.hpp
            │   ├── rnn_cell_decomposition.hpp
            │   ├── softplus_decomposition.hpp
            │   └── unroll_tensor_iterator.hpp
            ├── opset_conversions
            │   ├── convert_opset2_to_opset1.hpp
            │   └── convert_opset3_to_opset2.hpp
            ├── rt_info
            │   ├── fused_names_attribute.hpp
            │   └── primitives_priority_attribute.hpp
            ├── smart_reshape
            │   ├── reshape_with_hc_output.hpp
            │   ├── smart_reshape.hpp
            │   └── utils.hpp
            └── utils
                └── utils.hpp

```

### CODE REVIEW

File: inference-engine/src/legacy_api/CMakeLists.txt [hunk 1]
```diff
@@ -39,7 +39,7 @@ set_ie_threading_interface_for(${TARGET_NAME}_obj)
 target_compile_definitions(${TARGET_NAME}_obj PRIVATE IMPLEMENT_INFERENCE_ENGINE_API)
 
 target_include_directories(${TARGET_NAME}_obj PRIVATE
-    ${PUBLIC_HEADERS_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/src
+    ${PUBLIC_HEADERS_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/src ${CMAKE_CURRENT_SOURCE_DIR}/include/legacy/transformations
```
[the remark this answers is not in the document]
Comment (GlebKazantaev) [reply inference-engine/src/legacy_api/CMakeLists.txt:42]: will recheck

**Comment by ilya-lavrenov** [comment]
BTW, that is the difference between `common_transformations` and `common_optimizations` ?

**Comment by GlebKazantaev** [comment]
> BTW, that is the difference between `common_transformations` and `common_optimizations` ?

It is still an open for me where to keep common internal transformations

**Comment by ilya-lavrenov** [comment]
Looks like we need changes in other plugins and PR to product-config

**Review by slyalin — APPROVED** [review]

**Review by vladimir-paramuzov — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

**Review by ilyachur — APPROVED** [review]

**Pull request merged**